bentoml/_internal/models/model.py

```diff
@@ -57,24 +56,25 @@ class ModelSignatureDict(t.TypedDict, total=False):
 CUSTOM_OBJECTS_FILENAME = "custom_objects.pkl"
 
 
-if TYPE_CHECKING:
-    ModelOptionsSuper = UserDict[str, t.Any]
-else:
-    ModelOptionsSuper = UserDict
-
-
-class ModelOptions(ModelOptionsSuper):
-    @classmethod
-    def with_options(cls, **kwargs: t.Any) -> ModelOptions:
-        return cls(**kwargs)
+@attr.define
+class ModelOptions:
+    def with_options(self, **kwargs: t.Any) -> "ModelOptions":
+        return attr.evolve(self, **kwargs)
```

Comment:
I think it might be better to have these be fully abstract, or maybe have an ABC that people can subclass that's a supertype of this `attrs` base class.

Reply:
@ssheng are we still need to do something like this?

Reply:
Not all frameworks will have options. For those that don't have options, we can still have it fallback to using ModelOptions. So we probably don't want to change it to an abstract class.

Reply:
Some part of me likes the purity of having an ABC define the trait and then implementing it (with a default class) but it's probably realistically it's not important enough to add the code complexity for.